### DISCUSSION TITLE
Workload installation failed: Workload ID android/maui/ios isn't supported on this platform.

### DISCUSSION THREAD

**Discussion by thijs-philips** [unoplatform/uno#17992] [Q&A]
It feels more like a Visual Studio question than a Uno platform issue, but I cannot find a solution anywhere else. 
When I perform the uno-check on my system it fails at the last step:

```
fix failed - Workload Install failed: `dotnet workload install --from-rollback-file
"C:\TEMP\uno-check-xxxxx\workload.json" android ios maccatalyst macos maui --source
"https://api.nuget.org/v3/index.json"
```
when I manually execute this workload install, I get the marginally more descriptive error `Workload installation failed: Workload ID android isn't supported on this platform.` If I split it up, this happens for ios, maui and the other workloads as well. 

Is there any obvious reason for this behavior? I have been searching for an solution for hours, any help would be appreciated

I am running Windows 22H2 19045.4651, VS 17.11.1, Uno Check 1.25.1., and

```
:>dotnet --info
.NET SDK:
 Version:           8.0.303
 Commit:            29ab8e3268
 Workload version:  8.0.300-manifests.56cd0383
 MSBuild version:   17.10.4+10fbfbf2e

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19045
 OS Platform: Windows
 RID:         win-x86
 Base Path:   C:\Program Files (x86)\dotnet\sdk\8.0.303\

.NET workloads installed:
 [maui-windows]
   Installation Source: VS 17.11.35219.272
   Manifest Version:    8.0.72/8.0.100
   Manifest Path:       C:\Program Files (x86)\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.maui\8.0.72\WorkloadManifest.json
   Install Type:        FileBased


Host:
  Version:      8.0.8
  Architecture: x86
  Commit:       08338fcaa5

.NET SDKs installed:
  8.0.303 [C:\Program Files (x86)\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 5.0.1 [C:\Program Files (x86)\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.33 [C:\Program Files (x86)\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.7 [C:\Program Files (x86)\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.8 [C:\Program Files (x86)\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 5.0.1 [C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.29 [C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.33 [C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.20 [C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.7 [C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.8 [C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.29 [C:\Program Files (x86)\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.33 [C:\Program Files (x86)\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.20 [C:\Program Files (x86)\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.7 [C:\Program Files (x86)\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.8 [C:\Program Files (x86)\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x64   [C:\Program Files\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x64\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found
```

**Comment by thijs-philips** [comment]
I would be fine for now only building for Windows but no such luck:

I created an Uno app based on the template  and removed any build target except Windows and Desktop. Unfortunately this means that the Skia Resizetizer is failing
```
Uno.Resizetizer0000	System.TypeInitializationException: The type initializer for 'SkiaSharp.SKImageInfo' threw an exception. ---> System.BadImageFormatException: An attempt was made to load a program with an incorrect format. (Exception from HRESULT: 0x8007000B)
   at SkiaSharp.SkiaApi.sk_colortype_get_default_8888()
```
   
   My goal is to demonstrate the usefulness for using Uno as a way to make the Figma demonstrators in our company more realistic and interactive, but so far the beginning is a bit rough.

**Reply by francoistanguay** [reply]
Could you please confirm you're successfully run Uno Check and it outputs no errors? https://platform.uno/docs/articles/external/uno.check/doc/using-uno-check.html?tabs=windows#install-and-run-uno-check

Could you please also provide a binlog? https://platform.uno/docs/articles/uno-builds-troubleshooting.html#generating-msbuild-binary-log-files

Thank you, these things should help the team figure out what's going on with your setup.

**Reply by MartinZikmund** [reply]
@dansiegel Does the Resizetizer error above look familiar?

**Reply by jeromelaban** [reply]
This issue is likely related to https://github.com/unoplatform/uno.resizetizer/issues/263, for which some specific machine configuration is forcing 32 bits environment incorrectly in msbuild.

@thijs-philips make sure to provide the binlogs as mentioned above, thanks!

**Comment by thijs-philips** [comment]
Uno check did not run succesfully, it fails in the last step where it tries to install the workloads;  The error is as described above:

```
» Attempting to fix: .NET SDK - Workloads (8.0.300)
Fix failed - Workload Install failed: `dotnet workload install --from-rollback-file
"C:\...\workload.json" android ios maccatalyst macos maui --source
"https://api.nuget.org/v3/index.json"
```

I did follow the troubleshooting on that page, to no avail, I also did a full manual clean of all .Net SDKs, reinstalling only 8. but with the same error as result. 

The problem as I see it is this  (taking the Android workload as example):

The installation of the workloads fail
```  
C:\....\>dotnet workload install android
Workload installation failed: Workload ID android isn't supported on this platform. 
```

Because the workload is not recognized; In fact, the installable workload list is very empty:

```  
C:\...>dotnet workload search

Workload ID       Description
--------------------------------------------------------------------
macos             .NET SDK Workload for building macOS applications.
maui-tizen        .NET MAUI SDK for Tizen
maui-windows      .NET MAUI SDK for Windows
```

Even though the manifests (that, if I understand it correctly, let the system know which workloads can be installed) do seem to exist

```  
C:\...>dotnet workload update --ignore-failed-sources

Updated advertising manifest microsoft.net.workload.emscripten.net7.
Updated advertising manifest microsoft.net.sdk.aspire.
Updated advertising manifest microsoft.net.workload.emscripten.current.
Updated advertising manifest microsoft.net.sdk.maccatalyst.
Updated advertising manifest microsoft.net.workload.mono.toolchain.net6.
Updated advertising manifest microsoft.net.workload.mono.toolchain.net7.
Updated advertising manifest microsoft.net.workload.mono.toolchain.current.
Updated advertising manifest microsoft.net.workload.emscripten.net6.
Updated advertising manifest microsoft.net.sdk.android.
Updated advertising manifest microsoft.net.sdk.macos.
Updated advertising manifest microsoft.net.sdk.tvos.
Updated advertising manifest microsoft.net.sdk.ios.
Updated advertising manifest microsoft.net.sdk.maui.
```

I am happy to make a binlog, but I doubt if it will shed any additional light on the problem. Would you like me to?

**Reply by MartinZikmund** [reply]
I noticed you have Tizen workloads installed - I think they may be causing trouble with workload installation/updates - see https://github.com/Samsung/Tizen.NET/issues/223 . Using `--skip-manifest-update` could help here

**Reply by thijs-philips** [reply]
I tried it out, but it did not do anything, unfortunately. 

Looking at [https://github.com/Samsung/Tizen.NET/issues/223](https://github.com/Samsung/Tizen.NET/issues/223), it seems to be a little bit different from my issue. There the Tizen workloads were installed blocking others, in my case Tizen is in the list of manifests, but not even visible as an installable package.  Worth a shot, though. thanks!
